Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -460,9 +464,88 @@
     return WebCore::Cursor::Type::Pointer;
 }
 
-auto UnifiedPDFPlugin::pdfElementTypesForPluginPoint(const WebCore::IntPoint&) const -> PDFElementTypes
+WebCore::IntPoint UnifiedPDFPlugin::pointInDocumentSpaceForPluginPoint(const WebCore::IntPoint& point) const
 {
-    return { };
+    return WebCore::AffineTransform { }.scale(1.0f / m_documentLayout.scale()).translate(m_scrollOffset).mapPoint(point);
+}
+
+std::optional<PDFDocumentLayout::PageIndex> UnifiedPDFPlugin::nearestPageIndexForDocumentPoint(const WebCore::IntPoint& point) const
+{
+    std::optional<PDFDocumentLayout::PageIndex> nearestPageIndex;
+
+    switch (m_documentLayout.displayMode()) {
+    case PDFDocumentLayout::DisplayMode::TwoUp:
+    case PDFDocumentLayout::DisplayMode::TwoUpContinuous:
+        // FIXME: Implement page index resolution for TwoUp display modes.
+        notImplemented();
+        break;
+    case PDFDocumentLayout::DisplayMode::SinglePage:
+        // There is only one page to be near.
```

Comment:
I think smfr's plan for SinglePage is that they're actually all laid out in a row and we just use scroll-snap, so it's likely the Continuous code is the right way to go for this. (see also the comment in `PDFDocumentLayout::layoutPages`)

Reply:
Interpreted too literally -- will fallthrough on the `SinglePage` case now.

Reply:
Fixed in [af10ddc](https://github.com/WebKit/WebKit/commit/af10ddcd0e5fdcb907e4c6dc388e54411872b172)